specs/package/Package.md

```diff
@@ -0,0 +1,505 @@
+packages may have multiple locations and how to search across them. This puts a high bar on all
+developers using MSIX to do the legwork, and doing so correctly, consistently and efficiently. These
+new APIs provide a consistent, correct and easy to use solution.
+
+### 3.1.1. Package Location Search Order
+
+The search order for a package's locations is:
+
+1. External (User or Machine)
+2. Mutable
+3. Install
+
+**NOTE:** If a package has a UserExternal location then MachineExternal location is not checked
+(even if the package has one).
+
+### 3.1.2. FindPackageFileInPackage
+
+`FindPackageFileInPackage(packageFullName, filename[, options])` searches the location(s) of a specific package
+for the specified file, per the algorithm:
+
+```
+IF SearchUserExternalPath OR SearchMachineExternalPath in options
+    IF SearchUserExternalPath AND SearchMachineExternalPath in options
+        path = package.GetEffectiveExternalPath()
+    ELSEIF SearchUserExternalPath in options
+        path = package.GetUserExternalPath()
+    ELSEIF SearchMachineExternalPath in options
+        path = package.GetMachineExternalPath()     
+    IF path != null AND FileExists(path\filename)
+        RETURN path\filename       
+
+IF SearchMutablePath in options
+    path = package.GetMutablePath()
+    IF path != null AND FileExists(path\filename)
+        RETURN path\filename
+
+path = package.GetInstallPath()
+IF FileExists(path\filename)
+    RETURN path\filename
+
+// Not Found
+RETURN null
+```
+
+Overloads without the `options` parameter use the default (search everything).
+
+
+### 3.1.3. FindPackageFileInPackageGraph
+
+`FindPackageFileInPackageGraph(filename[, options])` searches the location(s) of each package in the caller's
+package graph for the specified file, per the algorithm:
+
+```
+FOREACH package IN GetPackageGraph():
+
+    file = FindPackageFileInPackage(package.fullname, filename)
+    IF file != null
+        RETURN file
+
+NEXT
+
+// Not found
+RETURN null
+```
+
+Overloads without the `options` parameter use the default (search everything).
+
+# 4. Examples
+
+## 4.1. FindPackageFileInPackage
+
+Locate `resource.pri` in the package `Contoso_1.2.3.4_x64__1234567890abc`.
+
+### 4.1.1. C# Example
+
+```c#
+using Microsoft.Windows.ApplicationModel;
+
+string GetResourcesPri(string packageFullName)
+{
+    var absoluteFilename = Package.FindPackageFileInPackage(packageFullName, "resources.pri");
+    if (absoluteFilename == null)
+    {
+        Console.WriteLine($"ERROR: resources.pri not found for {packageFullName}");
+        throw new FileNotFoundException($"resources.pri not found for {packageFullName}");
+    }
+    return absoluteFilename;
+}
+```
+
+### 4.1.2. C++ Example
+
+```c++
+wil::unique_process_heap_string GetResourcesPri(PCWSTR packageFullName)
+{
+    FindPackageFileOptions options{};
+    wil::unique_process_heap_string absoluteFilename;
```

Comment:
I don't understand why this is a unique process heap string.  It is a WinRT API so the `String` type in the .idl file should make this an `HSTRING`.  Therefore I'd expect this to be a `winrt::hstring` (or equivalent).